sql/core/src/main/scala/org/apache/spark/sql/execution/stat/StatFunctions.scala

```diff
@@ -77,4 +78,42 @@ private[sql] object StatFunctions {
       })
     counts.cov
   }
+
+  /** Generate a table of frequencies for the elements of two columns. */
+  private[sql] def crossTabulate(df: DataFrame, col1: String, col2: String): DataFrame = {
+    val tableName = s"${col1}_$col2"
+    val distinctCol2 = df.select(col2).distinct.collect().sortBy(_.get(0).toString)
```

Comment:
btw - isn't a more efficient way to run this is to do groupBy(col1, col2).count(), and then pivot the table?